drivers/modem/wncm14a2a.c

```diff
@@ -0,0 +1,1830 @@
+				     buf, sizeof(buf));
+	}
+#endif
+
+	SYS_LOG_ERR("Unknown IP address family:%d", addr->sa_family);
+	return NULL;
+}
+
+/* Send an AT command with a series of response handlers */
+static int send_at_cmd(struct wncm14a2a_socket *sock,
+			const u8_t *data, int timeout)
+{
+	int ret;
+
+	ictx.last_error = 0;
+
+	SYS_LOG_DBG("OUT: [%s]", data);
+	mdm_receiver_send(&ictx.mdm_ctx, data, strlen(data));
+	mdm_receiver_send(&ictx.mdm_ctx, "\r\n", 2);
+
+	if (timeout == K_NO_WAIT) {
+		return 0;
+	}
+
+	if (!sock) {
+		k_sem_reset(&ictx.response_sem);
+		ret = k_sem_take(&ictx.response_sem, timeout);
+	} else {
+		k_sem_reset(&sock->sock_send_sem);
+		ret = k_sem_take(&sock->sock_send_sem, timeout);
+	}
+
+	if (ret == 0) {
+		ret = ictx.last_error;
+	} else if (ret == -EAGAIN) {
+		ret = -ETIMEDOUT;
+	}
+
+	return ret;
+}
+
+static int send_data(struct wncm14a2a_socket *sock, struct net_pkt *pkt)
+{
+	int ret;
+	struct net_buf *frag;
+	char buf[sizeof("AT@SOCKWRITE=#,####,1\r")];
+
+	if (!sock) {
+		return -EINVAL;
+	}
+
+	ictx.last_error = 0;
+
+	frag = pkt->frags;
+	/* use SOCKWRITE with binary mode formatting */
+	snprintk(buf, sizeof(buf), "AT@SOCKWRITE=%d,%u,1%c",
+		 sock->socket_id, net_buf_frags_len(frag), '\r');
+	mdm_receiver_send(&ictx.mdm_ctx, buf, strlen(buf));
+
+	/* Loop through packet data and send */
+	while (frag) {
+		mdm_receiver_send(&ictx.mdm_ctx,
+				  frag->data, frag->len);
+		frag = frag->frags;
+	}
+
+	mdm_receiver_send(&ictx.mdm_ctx, "\r\n", 2);
+	k_sem_reset(&sock->sock_send_sem);
+	ret = k_sem_take(&sock->sock_send_sem, MDM_CMD_SEND_TIMEOUT);
+	if (ret == 0) {
+		ret = ictx.last_error;
+	} else if (ret == -EAGAIN) {
+		ret = -ETIMEDOUT;
+	}
+
+	return ret;
+}
+
+/*** NET_BUF HELPERS ***/
+
+static bool is_crlf(u8_t c)
+{
+	if (c == '\n' || c == '\r') {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+static void net_buf_skipcrlf(struct net_buf **buf)
+{
+	/* chop off any /n or /r */
+	while (*buf && is_crlf(*(*buf)->data)) {
+		net_buf_pull_u8(*buf);
+		if (!(*buf)->len) {
+			*buf = net_buf_frag_del(NULL, *buf);
+		}
+	}
+}
+
+static u16_t net_buf_findcrlf(struct net_buf *buf, struct net_buf **frag,
+			      u16_t *offset)
+{
+	u16_t len = 0, pos = 0;
+
+	while (buf && !is_crlf(*(buf->data + pos))) {
+		if (pos + 1 >= buf->len) {
+			len += buf->len;
+			buf = buf->frags;
+			pos = 0;
+		} else {
+			pos++;
+		}
+	}
+
+	if (buf && is_crlf(*(buf->data + pos))) {
+		len += pos;
+		*offset = pos;
+		*frag = buf;
+		return len;
+	}
+
+	return 0;
+}
+
+/*** UDP / TCP Helper Function ***/
+
+/* Setup IP header data to be used by some network applications.
+ * While much is dummy data, some fields such as dst, port and family are
+ * important.
+ * Return the IP + protocol header length.
+ */
+static int net_pkt_setup_ip_data(struct net_pkt *pkt,
+				  struct wncm14a2a_socket *sock)
+{
+	int hdr_len = 0;
+	u16_t src_port = 0, dst_port = 0;
+
+	if (net_pkt_family(pkt) == AF_INET6) {
```

Comment:
I think the original code was more readable when the ifdef was outside of the if statement. In net_context.c I did similar check like this

```
#if defined(CONFIG_NET_IPV6)
        if (net_pkt_family(pkt) == AF_INET6) {
        ...
        } else
#endif 
#if defined(CONFIG_NET_IPV4)
        if (net_pkt_family(pkt) == AF_INET4) {
        ...
        } else
#endif 
        {
            Return possible error here or leave empty if no error
        }
```

Reply:
Ok, I'll go through all of the #ifdefs like this one and flip the order again leaving a block at the end.